src/features/codeLensProvider.ts

```diff
@@ -29,19 +30,13 @@ export default class OmniSharpCodeLensProvider extends AbstractProvider implemen
 
     private _options: Options;
 
-    constructor(server: OmniSharpServer, testManager: TestManager) {
+    constructor(server: OmniSharpServer, testManager: TestManager, optionStream: OptionStream) {
         super(server);
 
-        this._resetCachedOptions();
-
-        let configChangedDisposable = vscode.workspace.onDidChangeConfiguration(this._resetCachedOptions, this);
+        let configChangedDisposable = optionStream.subscribe(options => this._options = options);
```

Comment:
Remove this, store field of optionStream, call optionStream.GetLatestOptions() where applicable